### PR TITLE
Build action support added

### PR DESCRIPTION
**Pull request by srids**

### CODE REVIEW

File: clusterdock/cli.py [hunk 1]
```diff
@@ -56,6 +56,21 @@ def main():
     build_parser = action_subparsers.add_parser('build',
                                                 formatter_class=FORMATTER_CLASS,
                                                 add_help=False)
+    build_parser.add_argument('--namespace',
+                              help='Namespace to use when looking for images',
```
Comment (dimaspivak) [inline clusterdock/cli.py]: Is this for looking for images or for naming them?

File: clusterdock/cli.py [hunk 2]
```diff
@@ -56,6 +56,21 @@ def main():
...
+                              metavar='ns')
+    build_parser.add_argument('--network',
+                              help='Docker network to use',
+                              default=defaults['DEFAULT_NETWORK'],
+                              metavar='nw')
+    build_parser.add_argument('-o', '--operating-system',
+                              help='Operating system to use for cluster nodes',
+                              metavar='sys')
+    build_parser.add_argument('-r', '--registry',
```
Comment (dimaspivak) [inline clusterdock/cli.py]: Same comment as above. To be honest, we may want to get rid of separate `--registry` and `--namespace` arguments and replace them to with `--repository` so people (or topologies) would just specify a repository once and for all and simplify a lot of stuff. What do you think? We'd need to also update topologies but I think it'd be worth it.

File: clusterdock/models.py [hunk 3]
```diff
@@ -365,6 +365,17 @@ def failure(timeout):
 
         self._add_node_to_etc_hosts()
 
+    def stop(self, remove=True):
+        """Stop the node by removing the Docker container.
```
Comment (dimaspivak) [inline clusterdock/models.py]: I think this should say `Stop the node and optionally removing the Docker container`.

File: clusterdock/utils.py [hunk 4]
```diff
@@ -95,3 +95,29 @@ def join_url_parts(*parts):
...
+def version_tuple(version, default=None):
+    """
+    Convert a version string or tuple to a tuple.
+    Will return (major, minor, release) kind of format.
+    """
+    if isinstance(version, str):
+        return tuple(int(x) for x in version.split('.'))
+    elif isinstance(version, tuple):
+        return version
+    else:
```
Comment (dimaspivak) [inline clusterdock/utils.py]: I think we can just leave out the ``default`` argument. I don't think people would need to change that default in this function very often and Python functions return `None` implicitly if we don't return anything explicit. Same comment for the function below.

File: setup.py [hunk 5]
```diff
@@ -23,6 +23,7 @@
     history = history_file.read()
 
 requirements = [
+    'colorlog',
```
Comment (dimaspivak) [inline setup.py]: This is fancy! :)

File: clusterdock/cli.py (hunk 1 above)
Comment (srids) [reply clusterdock/cli.py]: as discussed offline, removed this and now have 'repository' arg

File: clusterdock/cli.py [hunk 6]
```diff
@@ -56,20 +56,17 @@ def main():
     build_parser = action_subparsers.add_parser('build',
                                                 formatter_class=FORMATTER_CLASS,
                                                 add_help=False)
-    build_parser.add_argument('--namespace',
-                              help='Namespace to use when looking for images',
-                              metavar='ns')
+    build_parser.add_argument('--repository',
```
Comment (dimaspivak) [inline clusterdock/cli.py]: Nit: can you put this after `--operating-system` to keep it alphabetically ordered?

File: clusterdock/cli.py (hunk 2 above)
Comment (srids) [reply clusterdock/cli.py]: as discussed offline, removed 'registry' and 'namespace' args and now have 'repository' arg

File: clusterdock/models.py (hunk 3 above)
Comment (srids) [reply clusterdock/models.py]: Done

File: clusterdock/utils.py (hunk 4 above)
Comment (srids) [reply clusterdock/utils.py]: Done

**Pull request merged**